File: Ruby/foundation/JSONRequest.rb

require 'uri'
require 'net/http'
require 'cgi'
require 'response/HTTPResponse'

class JSONRequest

	def initialized
	  
	end
				
	def addQueryParameter(url, parameter, value)
		if (parameter!=nil) && (value!=nil) then
			if url.index('?')!=nil then
				url << '&'
			else
				url << '?'
			end
      url << CGI::escape(parameter.to_s)
      url << '='
      url << CGI::escape(value.to_s)
		end
		return url	
	end
	
  def get(url, responsetype, username, password)
    uri=URI.parse(url)
    req = Net::HTTP::Get.new(uri.request_uri)
    if responsetype!=nil then
      req["Accept"]="application/json"
    end
    req.basic_auth username, password
    response=HTTPResponse.new()
    begin
      httpresponse = Net::HTTP.new(uri.host, uri.port).start { |http| http.request(req) }
      response.setCode(httpresponse.code)
      response.setContent(httpresponse.body)
      response.setContentType(httpresponse.header['Content-Type'])
      response.setLocation(httpresponse.header['Location'])
    rescue => e
      print 'exception handled'
    end
    return response
  end
  
  def post(url, postdata, responsetype, username, password)
    uri=URI.parse(url)
    req = Net::HTTP::Post.new(uri.request_uri)
    if responsetype!=nil then
      req["Accept"]="application/json"
    end
    req.basic_auth username, password
    req.body=postdata
    response=HTTPResponse.new()
    begin
      httpresponse = Net::HTTP.new(uri.host, uri.port).start { |http| http.request(req) }
      response.setCode(httpresponse.code)
      response.setContent(httpresponse.body)
      response.setContentType(httpresponse.header['Content-Type'])
      response.setLocation(httpresponse.header['Location'])
    rescue => e
      print 'exception handled'
    end
    return response    
  end
 
  def postMultipart(url, postdata, responsetype, username, password, boundary)
    uri=URI.parse(url)
    req = Net::HTTP::Post.new(uri.request_uri)
    if responsetype!=nil then
      req["Accept"]="application/json"
    end
    
    req["Content-Type"]='multipart/mixed; boundary="'+boundary+'"';
      
    req.basic_auth username, password
    req.body=postdata
    response=HTTPResponse.new()
    begin
      httpresponse = Net::HTTP.new(uri.host, uri.port).start { |http| http.request(req) }
      response.setCode(httpresponse.code)
      response.setContent(httpresponse.body)
      response.setContentType(httpresponse.header['Content-Type'])
      response.setLocation(httpresponse.header['Location'])
    rescue => e
      print 'exception handled'
    end
    return response    
  end
 
  def delete(url, username, password)
    uri=URI.parse(url)
    req = Net::HTTP::Delete.new(uri.request_uri)
    req.basic_auth username, password
    response=HTTPResponse.new()
    begin
      httpresponse = Net::HTTP.new(uri.host, uri.port).start { |http| http.request(req) }
      response.setCode(httpresponse.code)
      response.setContent(httpresponse.body)
      response.setContentType(httpresponse.header['Content-Type'])
      response.setLocation(httpresponse.header['Location'])
    rescue => e
      print 'exception handled'
    end
    return response
  end

end
		

File: Python/foundation/JSONRequest.py

import urllib
import urllib2
import base64

from response.HTTPResponse import HTTPResponse
from urllib2 import HTTPError

class JSONRequest:

	def __init__(self):		
		self.initialised=True
				
	def addQueryParameter(self, url, parameter, value):
		if parameter is not None and value is not None:
			if '?' in url:
				url=url+'&'
			else:
				url=url+'?'
			url=url+urllib.urlencode({parameter : value})
		return url	
				
	def get(self, url, responsetype, username, password):
		opener = urllib2.build_opener(urllib2.HTTPHandler)
		request = urllib2.Request(url)
		if responsetype is not None:
			request.add_header('Accept', responsetype)
		request.get_method = lambda: 'GET'
		base64string = base64.encodestring('%s:%s' % (username,password))[:-1]
		request.add_header("Authorization", "Basic %s" % base64string)
		response=HTTPResponse(None)
		try:
			handle = opener.open(request)
			info=handle.info()
			content=handle.read()
			response.setCode(handle.getcode())
			response.setContent(content)
			print 'Response code = %d' % (handle.getcode())
			if info is not None:
				headers=info.dict
				if headers is not None:
					if 'location' in headers:
						response.setLocation(headers['location'])
					if 'content-type' in headers:
						response.setContentType(headers['content-type'])
		except HTTPError, e:
			print 'The server couldn\'t fulfill the request.'
			print 'Error code: ', e.code			
			response.setCode(e.code)
		return response

	def post(self, url, postdata, responsetype, username, password):
		opener = urllib2.build_opener(urllib2.HTTPHandler)
		request = urllib2.Request(url, postdata)
		if responsetype is not None:
			request.add_header('Accept', responsetype)
		request.get_method = lambda: 'POST'
		base64string = base64.encodestring('%s:%s' % (username,password))[:-1]
		request.add_header("Authorization", "Basic %s" % base64string)
		response=HTTPResponse(None)
		try:
			handle = opener.open(request)
			info=handle.info()
			content=handle.read()
			response.setCode(handle.getcode())
			response.setContent(content)
			print 'Response code = %d' % (handle.getcode())
			if info is not None:
				headers=info.dict
				if headers is not None:
					if 'location' in headers:
						response.setLocation(headers['location'])
					if 'content-type' in headers:
						response.setContentType(headers['content-type'])
		except HTTPError, e:
			print 'The server couldn\'t fulfill the request.'
			print 'Error code: ', e.code			
			response.setCode(e.code)
		return response

	def postMultipart(self, url, postdata, responsetype, username, password, boundary):
		opener = urllib2.build_opener(urllib2.HTTPHandler)
		request = urllib2.Request(url, postdata)
		if responsetype is not None:
			request.add_header('Accept', responsetype)
		request.get_method = lambda: 'POST'
		base64string = base64.encodestring('%s:%s' % (username,password))[:-1]
		request.add_header("Authorization", "Basic %s" % base64string)
		request.add_header("Content-Type", 'multipart/mixed; boundary="'+boundary+'"')
		response=HTTPResponse(None)
		try:
			handle = opener.open(request)
			info=handle.info()
			content=handle.read()
			response.setCode(handle.getcode())
			response.setContent(content)
			print 'Response code = %d' % (handle.getcode())
			if info is not None:
				headers=info.dict
				if headers is not None:
					if 'location' in headers:
						response.setLocation(headers['location'])
					if 'content-type' in headers:
						response.setContentType(headers['content-type'])
		except HTTPError, e:
			print 'The server couldn\'t fulfill the request.'
			print 'Error code: ', e.code			
			response.setCode(e.code)
		return response

	def delete(self, url, username, password):
		opener = urllib2.build_opener(urllib2.HTTPHandler)
		request = urllib2.Request(url)
		request.get_method = lambda: 'DELETE'
		base64string = base64.encodestring('%s:%s' % (username,password))[:-1]
		request.add_header("Authorization", "Basic %s" % base64string)
		response=HTTPResponse(None)
		try:
			handle = opener.open(request)
			info=handle.info()
			response.setCode(handle.getcode())
			print 'Response code = %d' % (handle.getcode())
			if info is not None:
				headers=info.dict
				if headers is not None:
					if 'location' in headers:
						response.setLocation(headers['location'])
					if 'content-type' in headers:
						response.setContentType(headers['content-type'])
		except HTTPError, e:
			print 'The server couldn\'t fulfill the request.'
			print 'Error code: ', e.code			
			response.setCode(e.code)
		return response

		